src/app/parcels/getExpandedParcelDetails.ts

```diff
@@ -47,35 +72,56 @@ export const getRawParcelDetails = async (parcelId: string) => {
         )
         .eq("primary_key", parcelId)
         .single();
-    if (error) {
-        const logId = await logErrorReturnLogId("Error with fetch: Parcel", error);
-        throw new DatabaseError("fetch", "parcel", logId);
-    }
-    return data;
-};
-
-export const familyCountToFamilyCategory = (count: number): string => {
-    if (count <= 1) {
-        return "Single";
-    }
 
-    if (count <= 9) {
-        return `Family of ${count}`;
+    if (error) {
+        const logId = await logErrorReturnLogId("Failed to fetch expanded parcel details", {
+            error,
+        });
+        return {
+            parcelDetails: null,
+            error: {
+                type: "failedToFetchParcelDetails",
+                logId,
+            },
+        };
     }
 
-    return "Family of 10+";
-};
+    const client = rawParcelDetails.client;
 
-export const formatDatetimeAsDate = (datetime: Date | string | null): string => {
-    if (datetime instanceof Date) {
-        return datetime.toLocaleDateString("en-GB");
-    }
+    if (client === null) {
+        const logId = await logErrorReturnLogId(
+            "Failed to fetch client details when fetching expanded parcel details",
+            { dbParcelDetails: rawParcelDetails }
+        );
 
-    if (datetime === null || isNaN(Date.parse(datetime))) {
-        return "-";
+        return {
+            parcelDetails: null,
+            error: {
+                type: "clientDetailDoesNotExist",
+                logId,
+            },
+        };
     }
 
-    return new Date(datetime).toLocaleDateString("en-GB");
+    return {
+        parcelDetails: {
+            expandedParcelData: {
+                voucherNumber: rawParcelDetails.voucher_number ?? "",
```

Comment:
is it possible for rawParcelDetails to be null? if so should we return a specific error?

Reply:
Okay cool!